### PR TITLE
Fix globals, part 10: `atob()` and `btoa()`

### PR DESCRIPTION
**Pull request by skyclouds2001**
### Related issues and pull requests

part of the https://github.com/mdn/content/issues/33019

### CODE REVIEW

File: files/en-us/web/api/workerglobalscope/atob/index.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+---
```
Comment (wbamberg) [inline files/en-us/web/api/workerglobalscope/atob/index.md:1]: I'm going to have the same comments for this and btoa that I had for the Window version.

**Review by wbamberg — CHANGES_REQUESTED** [review]
Thank you for your ongoing work on this project!

File: files/en-us/glossary/base64/index.md [hunk 2]
```diff
@@ -94,8 +94,8 @@ await dataUrlToBytes("data:application/octet-stream;base64,AAEC"); // Uint8Array
 ## See Also
 
 - JavaScript APIs:
-  - [btoa() global function](/en-US/docs/Web/API/btoa)
-  - [atob() global function](/en-US/docs/Web/API/atob)
+  - {{domxref("Window.atob()")}} and {{domxref("WorkerGlobalScope.atob()")}}
+  - {{domxref("Window.btoa()")}} and {{domxref("WorkerGlobalScope.btoa()")}}
```
Comment (skyclouds2001) [inline files/en-us/glossary/base64/index.md:98]:
```suggestion
  - {{domxref("Window.atob()")}} (also {{domxref("WorkerGlobalScope.atob(), "available in workers", "", "nocode")}})
  - {{domxref("Window.btoa()")}} (also {{domxref("WorkerGlobalScope.btoa(), "available in workers", "", "nocode")}})
```

File: files/en-us/glossary/base64/index.md [hunk 3]
```diff
@@ -33,8 +33,8 @@ This means that the Base64 version of a string or file is typically roughly a th
 
 Browsers natively provide two JavaScript functions for decoding and encoding Base64 strings:
 
-- [`btoa`](/en-US/docs/Web/API/btoa): creates a Base64-encoded ASCII string from a string of binary data ("btoa" should be read as "binary to ASCII").
-- [`atob`](/en-US/docs/Web/API/atob): decodes a Base64-encoded string ("atob" should be read as "ASCII to binary").
+- {{domxref("Window.btoa()")}} and {{domxref("WorkerGlobalScope.btoa()")}}: creates a Base64-encoded ASCII string from a string of binary data ("btoa" should be read as "binary to ASCII").
+- {{domxref("Window.atob()")}} and {{domxref("WorkerGlobalScope.atob()")}}: decodes a Base64-encoded string ("atob" should be read as "ASCII to binary").
```
Comment (skyclouds2001) [inline files/en-us/glossary/base64/index.md:37]:
```suggestion
- {{domxref("Window.btoa()")}} (also {{domxref("WorkerGlobalScope.btoa(), "available in workers", "", "nocode")}}): creates a Base64-encoded ASCII string from a string of binary data ("btoa" should be read as "binary to ASCII").
- {{domxref("Window.atob()")}} (also {{domxref("WorkerGlobalScope.atob(), "available in workers", "", "nocode")}}): decodes a Base64-encoded string ("atob" should be read as "ASCII to binary").
```

File: files/en-us/web/api/workerglobalscope/atob/index.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+## Browser compatibility
+
+{{Compat}}
+
+## See also
+
+- [A polyfill of `atob`](https://github.com/zloirock/core-js#base64-utility-methods) is available in [`core-js`](https://github.com/zloirock/core-js)
+- [`data` URLs](/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URLs)
+- {{domxref("Window.atob()")}}
```
Comment (skyclouds2001) [inline files/en-us/web/api/workerglobalscope/atob/index.md:59]:
```suggestion
- {{domxref("Window.atob()")}}: the same method, but in window scopes.
```

File: files/en-us/web/api/workerglobalscope/btoa/index.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+## Browser compatibility
+
+{{Compat}}
+
+## See also
+
+- [A polyfill of `btoa`](https://github.com/zloirock/core-js#base64-utility-methods) is available in [`core-js`](https://github.com/zloirock/core-js)
+- [`data` URLs](/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URLs)
+- {{domxref("WorkerGlobalScope.btoa()")}}
+- {{domxref("Window.atob()")}}
```
Comment (skyclouds2001) [inline files/en-us/web/api/workerglobalscope/btoa/index.md:80]:
```suggestion
- {{domxref("WorkerGlobalScope.atob()")}}
- {{domxref("Window.btoa()")}}: the same method, but in window scopes.
```

File: files/en-us/web/api/workerglobalscope/atob/index.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+{{APIRef("HTML DOM")}}
+
+The **`atob()`** method of the {{domxref("WorkerGlobalScope")}} interface decodes a
+string of data which has been encoded using {{glossary("Base64")}} encoding. You can use
+the {{domxref("WorkerGlobalScope.btoa()")}} method to encode and transmit
+data which may otherwise cause communication problems, then transmit it and use the
+`atob()` method to decode the data again. For example, you can encode,
+transmit, and decode control characters such as {{Glossary("ASCII")}} values 0 through 31.
+
+For use with arbitrary Unicode strings, see _The "Unicode Problem"_ section of the {{Glossary("Base64")}} glossary entry.
```
Comment (skyclouds2001) [inline files/en-us/web/api/workerglobalscope/atob/index.md:18]:
```suggestion
For use with arbitrary Unicode strings, see _The "Unicode Problem"_ section in {{Glossary("Base64")}} glossary entry.
```

File: files/en-us/web/api/workerglobalscope/btoa/index.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+console.log(ok.codePointAt(0).toString(16)); //   61: occupies < 1 byte
+
+const notOK = "✓";
+console.log(notOK.codePointAt(0).toString(16)); // 2713: occupies > 1 byte
+
+console.log(self.btoa(ok)); // YQ==
+console.log(self.btoa(notOK)); // error
+```
+
+For how to work around this limitation when dealing with arbitrary Unicode text, see _The "Unicode Problem"_ section of the {{Glossary("Base64")}} glossary entry.
```
Comment (skyclouds2001) [inline files/en-us/web/api/workerglobalscope/btoa/index.md:65]:
```suggestion
For how to work around this limitation when dealing with arbitrary Unicode text, see _The "Unicode Problem"_ in the {{Glossary("Base64")}} glossary entry.
```

File: files/en-us/glossary/base64/index.md [hunk 8]
```diff
@@ -94,8 +94,8 @@ await dataUrlToBytes("data:application/octet-stream;base64,AAEC"); // Uint8Array
 ## See Also
 
 - JavaScript APIs:
-  - [btoa() global function](/en-US/docs/Web/API/btoa)
-  - [atob() global function](/en-US/docs/Web/API/atob)
+  - {{domxref("Window.atob()")}} (also {{domxref("WorkerGlobalScope.atob(), "available in workers", "", "nocode")}})
+  - {{domxref("Window.btoa()")}} (also {{domxref("WorkerGlobalScope.btoa(), "available in workers", "", "nocode")}})
```
Comment (skyclouds2001) [inline files/en-us/glossary/base64/index.md:98]:
```suggestion
  - {{domxref("Window.atob()")}} (also {{domxref("WorkerGlobalScope.atob()", "available in workers", "", "nocode")}})
  - {{domxref("Window.btoa()")}} (also {{domxref("WorkerGlobalScope.btoa()", "available in workers", "", "nocode")}})
```

File: files/en-us/glossary/base64/index.md [hunk 9]
```diff
@@ -33,8 +33,8 @@ This means that the Base64 version of a string or file is typically roughly a th
 
 Browsers natively provide two JavaScript functions for decoding and encoding Base64 strings:
 
-- [`btoa`](/en-US/docs/Web/API/btoa): creates a Base64-encoded ASCII string from a string of binary data ("btoa" should be read as "binary to ASCII").
-- [`atob`](/en-US/docs/Web/API/atob): decodes a Base64-encoded string ("atob" should be read as "ASCII to binary").
+- {{domxref("Window.btoa()")}} (also {{domxref("WorkerGlobalScope.btoa(), "available in workers", "", "nocode")}}): creates a Base64-encoded ASCII string from a string of binary data ("btoa" should be read as "binary to ASCII").
+- {{domxref("Window.atob()")}} (also {{domxref("WorkerGlobalScope.atob(), "available in workers", "", "nocode")}}): decodes a Base64-encoded string ("atob" should be read as "ASCII to binary").
```
Comment (skyclouds2001) [inline files/en-us/glossary/base64/index.md:37]:
```suggestion
- {{domxref("Window.btoa()")}} (also {{domxref("WorkerGlobalScope.btoa()", "available in workers", "", "nocode")}}): creates a Base64-encoded ASCII string from a string of binary data ("btoa" should be read as "binary to ASCII").
- {{domxref("Window.atob()")}} (also {{domxref("WorkerGlobalScope.atob()", "available in workers", "", "nocode")}}): decodes a Base64-encoded string ("atob" should be read as "ASCII to binary").
```

**Review by wbamberg — APPROVED** [review]
👍 thank you  @skyclouds2001 !

**Pull request merged**